app/models/concerns/auditable.rb

```diff
@@ -8,12 +8,21 @@ module Auditable
 
     def audit!(activity_type:, activity_information: nil, audit_comment: nil)
       audits.create!(
-        user: Current.user,
+        user: user,
         activity_type: activity_type,
         activity_information: activity_information,
         audit_comment: audit_comment,
-        api_user: Current.api_user
+        api_user: api_user,
+        automated_activity: user.blank? && api_user.blank?
```

Comment:
At this point if there's no current user or api user we know it's a an automated action.